.github/composite_actions/log_metric/action.yaml

```diff
@@ -0,0 +1,44 @@
+name: Log Metric
+description: Log data point to a metric with the provided value. If the metric is not there, it will create one
+# To avoid 'Credentials could not be loaded' calling workflows must include:
+# permissions:
+#  id-token: write
+#  contents: read
+inputs:
+  aws-region:
+    required: true
+    description: The AWS region
+  role-to-assume:
+    required: true
+    description: The role to assume in the STS session
+  metric-name:
+    description: Name of the metric to track in Cloudwatch.
+    required: true
+  value:
+    # Why we publish value 0 on success: https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/publishingMetrics.html#publishingZero
+    description: Value of the metric to track in Cloudwatch.
+    required: true
+  dimensions:
+    description: Dimensions of metric to track in Cloudwatch, in format dimensionName1=value,dimensionName2=value,...
+    required: false
+runs:
+  using: 'composite'
+  steps:
+    - name: Configure AWS credentials
+      uses: aws-actions/configure-aws-credentials@67fbcbb121271f7775d2e7715933280b06314838 # 1.7.0
+      with:
+        role-to-assume: ${{ inputs.role-to-assume }}
+        aws-region: ${{ inputs.aws-region }}
+        role-duration-seconds: 900
+
+    - name: Setup Dart
```

Comment:
nit/question: Can we skip this as used here bc used in a context where flutter already installed?

Reply:
yes not necessary, good call.